superset-frontend/packages/superset-ui-core/src/color/utils.ts

```diff
@@ -51,3 +52,22 @@ export function getContrastingColor(color: string, thresholds = 186) {
 
   return r * 0.299 + g * 0.587 + b * 0.114 > thresholds ? '#000' : '#FFF';
 }
+
+export function getAnalogousColors(colors: string[], results: number) {
+  const generatedColors: string[] = [];
+  const ext = 3;
```

Comment:
I'm curious how we landed on 3 for this?

Reply:
A comment might be useful for future readers.

Reply:
Good catch! This is to solve the problem that the first three values generated by `tinycolor.analogous` may have the same or very close colors. You can see my local test:
<img width="1427" alt="image" src="https://user-images.githubusercontent.com/11830681/160428321-5f9c5ab3-c653-42db-b2f0-2c749bec9975.png">
Let me add some comments here.